lib/puppet/provider/cert/katello_ssl_tool.rb

```diff
@@ -15,7 +15,11 @@ def generate!
       FileUtils.mkdir_p(build_path)
       FileUtils.cp(resource[:custom_pubkey], build_path(File.basename(pubkey)))
       FileUtils.cp(resource[:custom_privkey], build_path(File.basename(privkey)))
-      FileUtils.cp(resource[:custom_req], build_path(File.basename(req_file)))
+      if resource[:custom_req]
+        FileUtils.cp(resource[:custom_req], build_path(File.basename(req_file)))
+      else
+        File.open(build_path(File.basename(req_file)), 'w') { |f| f.write('') }
```

Comment:
Seems a smidge ugly to write an empty file. Would it not be more appropriate to fix our SSL tool so this parameter is not required?

Reply:
:hammer: but this is so much easier :)

Reply:
@ehelms yes it would, but I'm afraid that's opening such a can of worms that I'll spend days to learn it. On the other hand it might be needed at some point, depending on where we want to go in the long term.